### PR TITLE
removed reactive components

### PR DESCRIPTION
**Pull request by samiwh**
Removing reactive components

### CODE REVIEW

File: workers/unity/Packages/com.improbable.gdk.health/Systems/HealthRegenSystem.cs [hunk 1]
```diff
@@ -16,38 +15,45 @@ public struct EntitiesNeedingRegenData
...
         [Inject] private EntitiesToRegen toRegen;
+        [Inject] private CommandSystem commandSystem;
+        [Inject] private ComponentUpdateSystem componentUpdateSystem;
 
         protected override void OnUpdate()
         {
             // Add the HealthRegenData if you don't currently have it.
             for (var i = 0; i < needData.Length; i++)
             {
+                if (!needData.Authority[i].HasAuthority)
```
Comment (zeroZshadow) [inline workers/unity/Packages/com.improbable.gdk.health/Systems/HealthRegenSystem.cs]: This should use SetFilter instead.
This require's us to no longer `Inject` structs but to make proper groups, which is the future API post Entities.26 anyways.

File: workers/unity/Packages/com.improbable.gdk.health/Systems/HealthRegenSystem.cs [hunk 2]
```diff
@@ -64,18 +70,25 @@ protected override void OnUpdate()
...
-                    var modifier = modifiedEvent.Modifier;
+                    var modifier = events[j].Event.Payload.Modifier;
                     if (modifier.Amount < 0)
                     {
                         damagedRecently = true;
                         break;
                     }
                 }
+
+
```
Comment (zeroZshadow) [inline workers/unity/Packages/com.improbable.gdk.health/Systems/HealthRegenSystem.cs]: nite: whitespace

File: workers/unity/Packages/com.improbable.gdk.health/Systems/HealthRegenSystem.cs [hunk 3]
```diff
@@ -146,14 +164,13 @@ protected override void OnUpdate()
                         regenData.NextRegenTimer += regenComponent.RegenInterval;
 
                         // Send command to regen entity.
-                        var commandSender = toRegen.ModifyHealthCommandSenders[i];
                         var modifyHealthRequest = new HealthComponent.ModifyHealth.Request(
                             toRegen.EntityId[i].EntityId,
                             new HealthModifier()
                             {
-                                Amount = regenComponent.RegenAmount
+                                Amount = regenComponent.RegenAmount - 3
```
Comment (zeroZshadow) [inline workers/unity/Packages/com.improbable.gdk.health/Systems/HealthRegenSystem.cs]: temp hack should be removed

File: workers/unity/Packages/com.improbable.gdk.health/Systems/ServerHealthModifierSystem.cs [hunk 4]
```diff
@@ -6,57 +6,48 @@
...
-            for (var i = 0; i < entitiesWithModifiedHealth.Length; i++)
+            var requests = commandSystem.GetRequests<HealthComponent.ModifyHealth.ReceivedRequest>();
+            for (int i = 0; i < requests.Count; ++i)
             {
-                var health = entitiesWithModifiedHealth.Health[i];
-                var healthModifiedEventSender = entitiesWithModifiedHealth.HealthModifiedEventSenders[i];
+                var entityId = requests[i].EntityId;
+                ref readonly var modifier = ref requests[i].Payload;
+                workerSystem.TryGetEntity(entityId, out var entity);
+                var health = EntityManager.GetComponentData<HealthComponent.Component>(entity);
```
Comment (zeroZshadow) [inline workers/unity/Packages/com.improbable.gdk.health/Systems/ServerHealthModifierSystem.cs]: This should be a `GetComponentDataFromEntity` outside the loop.

File: workers/unity/Packages/com.improbable.gdk.health/Systems/ServerHealthModifierSystem.cs [hunk 5]
```diff
@@ -6,57 +6,48 @@
...
-                    healthModifiedEventSender.Events.Add(healthModifiedInfo);
+                if (health.Health <= 0)
+                {
+                    healthModifiedInfo.Died = true;
                 }
 
-                entitiesWithModifiedHealth.Health[i] = health;
+                updateSystem.SendEvent(new HealthComponent.HealthModified.Event(healthModifiedInfo), entityId);
+
+                EntityManager.SetComponentData(entity, health);
```
Comment (zeroZshadow) [inline workers/unity/Packages/com.improbable.gdk.health/Systems/ServerHealthModifierSystem.cs]: See comment on `GetComponentDataFromEntity`

**Comment by zeroZshadow** [comment]
Just made some quick comments on ECS API usage which could be improved.

File: workers/unity/Packages/com.improbable.gdk.health/Systems/HealthRegenSystem.cs (hunk 3 above)
Comment (samiwh) [reply workers/unity/Packages/com.improbable.gdk.health/Systems/HealthRegenSystem.cs]: This PR is a temporary hack to test performance

File: workers/unity/Packages/com.improbable.gdk.health/Systems/HealthRegenSystem.cs (hunk 1 above)
Comment (samiwh) [reply workers/unity/Packages/com.improbable.gdk.health/Systems/HealthRegenSystem.cs]: This was the minimal change needed to test the difference

**Pull request merged**